src/components/FormAlertWrapper.js

```diff
@@ -0,0 +1,115 @@
+import compose from '../libs/compose';
+import OfflineIndicator from './OfflineIndicator';
+import networkPropTypes from './networkPropTypes';
+import styles from '../styles/styles';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+
+const propTypes = {
+    /** Wrapped child components */
+    children: PropTypes.node.isRequired,
+
+    /** Styles for container element */
+    containerStyles: PropTypes.arrayOf(PropTypes.object),
+
+    /** Whether to show the alert text */
+    isAlertVisible: PropTypes.bool.isRequired,
+
+    /** Whether message is in html format */
+    isMessageHtml: PropTypes.bool,
+
+    /** Error message to display above button */
+    message: PropTypes.string,
+
+    /** Props to detect online status */
+    network: networkPropTypes.isRequired,
+
+    /** Callback fired when the "fix the errors" link is pressed */
+    onFixTheErrorsLinkPressed: PropTypes.func,
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    containerStyles: [],
+    isMessageHtml: false,
+    message: '',
+    onFixTheErrorsLinkPressed: () => {},
+};
+
+// The FormAlertWrapper offers a standardized way of showing error messages and offline functionality.
+//
+// This component takes other components as a child prop. It will then render any wrapped components as a function using "render props",
+// and passes it a (bool) isOffline parameter. Child components can then use that isOffline to determine offline behavior.
+const FormAlertWrapper = (props) => {
+    function getAlertPrompt() {
+        let error = '';
+
+        if (!_.isEmpty(props.message)) {
+            if (props.isMessageHtml) {
+                error = (
+                    <RenderHTML html={`<muted-text>${props.message}</muted-text>`} />
+                );
+            } else {
+                error = (
+                    <Text style={styles.mutedTextLabel}>{props.message}</Text>
+                );
+            }
+        } else {
+            error = (
+                <>
+                    <Text style={styles.mutedTextLabel}>
+                        {`${props.translate('common.please')} `}
+                    </Text>
+                    <TextLink
+                        style={styles.label}
+                        onPress={props.onFixTheErrorsLinkPressed}
+                    >
+                        {props.translate('common.fixTheErrors')}
+                    </TextLink>
+                    <Text style={styles.mutedTextLabel}>
+                        {` ${props.translate('common.inTheFormBeforeContinuing')}.`}
+                    </Text>
+                </>
+            );
+        }
+
+        return (
+            <View style={[styles.flexRow, styles.ml2, styles.flexWrap, styles.flex1]}>
+                {error}
+            </View>
+        );
+    }
+
+    return (
+        <View style={[styles.mh5, styles.mb5, styles.flex1, styles.justifyContentEnd, ...props.containerStyles]}>
+            {props.isAlertVisible && (
```

Comment:
Why is `isAlertVisible` here instead of doing an early return at the top of `render()` and returning `null`? If the alert isn't supposed to be visible, why render the container with nothing in it?